### PR TITLE
Added ARGA Model

### PR DESCRIPTION
**Pull request by jlevy44**
On a rush out the door for a trip, so need help formatting and making topology generalizable. Minibatch support needed.

I'm sure there are a lot of improvements needed. Feel free to comment/edit where necessary to format to the specifications of this repo.

 When we PR this, I will add ARGVA, then graphVAE.

### CODE REVIEW

**Comment by jlevy44** [comment]
We should also find a nice test dataset. I'm not getting good results on my set.

**Comment by jlevy44** [comment]
#98

**Comment by rusty1s** [comment]
Hi, thanks for your code. I added `model/autoencoder.py` and `examples/autoencoder.py` files in which I implemented the [GAE](https://arxiv.org/pdf/1611.07308.pdf) based on your code. I think we can add VGAE and ARGA by inheriting from `GAE`. What do you think?

Cora results:
* AUC: 90.9
* AP: 91.8

**Comment by rusty1s** [comment]
Replacing the BCELoss with a custom loss op gives huge speed-ups.

**Comment by jlevy44** [comment]
Awesome, thanks. I’ll run some tests on my datasets and submit some PRs for VGAE and ARGA

**Comment by jlevy44** [comment]
I no longer see the GAE implementation. Can you point me to it? I’m going to work on a new PR.

File: examples/autoencoder.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+class Encoder(torch.nn.Module):
+    def __init__(self, in_channels, out_channels):
+        super(Encoder, self).__init__()
+        self.conv1 = GCNConv(in_channels, 2 * out_channels)
+        self.conv2 = GCNConv(2 * out_channels, out_channels)
+
+        self.lin1 = torch.nn.Linear(in_channels, 16)
+
+    def forward(self, x, edge_index):
+        return self.lin1(x)
```
Comment (jlevy44) [inline examples/autoencoder.py]: Why are we skipping the GCN Layers?

**Comment by jlevy44** [comment]
I can test with my small dataset before we merge in any case. Seems like the Cora test is good enough though.

**Comment by rusty1s** [comment]
Oh, I pushed the wrong encoder. Thanks for finding this. I will merge and correct it in master.

**Comment by rusty1s** [comment]
Fixed in master.
